lib/net/http/response.rb

```diff
@@ -224,13 +224,28 @@ def initialize(httpv, code, msg)   #:nodoc: internal use only
   # Accept-Encoding header from the user.
   attr_accessor :decode_content
 
-  # The encoding to use for the response body. If Encoding, use that encoding.
-  # If other true value, attempt to detect the appropriate encoding, and use
-  # that.
+  # Returns the value set by body_encoding=, or +false+ if none;
+  # see #body_encoding=.
   attr_reader :body_encoding
 
-  # Set the encoding to use for the response body.  If given a String, find
-  # the related Encoding.
+  # Sets the encoding that should be used when reading the body:
+  #
+  # - If the given value is an Encoding object, that encoding will be used.
+  # - Otherwise if the value is a string, the value of
+  #   {Encoding#find(value)}[https://docs.ruby-lang.org/en/master/Encoding.html#method-c-find]
+  #   will be used.
+  # - Otherwise an encoding will be deduced from the body itself.
+  #
+  # Examples:
+  #
+  #   res = Net::HTTP.get_response(hostname, '/')
+  #   # => #<Net::HTTPOK 200 OK readbody=true>
+  #   res.body_encoding # => false
+  #   res.body_encoding = Encoding::US_ASCII
+  #   res.body_encoding # => #<Encoding:US-ASCII>
+  #   res.body_encoding = 'US-ASCII'
+  #   res.body_encoding # => #<Encoding:US-ASCII>
```

Comment:
This example doesn't really show what the purpose of this method is. It just show that it is a setter.

Reply:
Not sure what you mean.  Suggestion?

Reply:
What I mean is that this example isn't showing the purpose of the `#body_encoding=`.

An example like this shows the purpose:

```ruby
http = Net::HTTP.new(hostname)
req = Net::HTTP::Get.new('/')

http.request(req) do |res|
  p res.body.encoding # => #<Encoding:ASCII-8BIT>
end

http.request(req) do |res|
  res.body_encoding = "UTF-8"
  p res.body.encoding # => #<Encoding:UTF-8>
end

Reply:
Substituted in, and thanks.